homework.md

```diff
@@ -44,70 +44,73 @@ Using CSS, how would I select every `<h1>` tag on the page?
 
 What is JavaScript?
 
-A programming language used on the internet
+Javascript is a scripting language that enables you to create dynamically updating content, control multimedia, animate images. It's a dynamic programming language that's used for web development, in web applications, for game development, and lots more.
 
 ## Question 8
 
 What three problem does Git & Github solve?
 
-When people want to show off code to each other they can put it on Github
+GitHub is a website for developers and programmers to collaboratively work on code.
+
+The primary benefit of GitHub is its version control system, which allows for seamless collaboration without compromising the integrity of the original project. 
+
+It's a great place to host your portfolio whilst demonstrating the ability to use and understand it. 
 
 ## Question 9
 
 What happens when you `fork` a repository?
 
-You delete it
+You make a copy of it on your own remote repo branch, so you can freely experiment with changes without affecting the original project.
 
 ## Question 10 
 
-What happens when you clone a repostory?
+What happens when you clone a repository?
 
-It send it to a friend
+You download a copy of the remote repo to your local machine allowing you to work on the files locally rather than directly with the source files of the origin repo. 
 
 ## Question 11
 
 What is a Pull Request?
 
-When you send a file over the internet
+Pull requests tell others about changes you've pushed to a branch in a repo. It notifies colleagues that you have work to be discussed and reviewed. 
 
 ## Question 12
 
 What is the Git History log?
 
-It lets you see into the future
+It lists all the commits made in a repo in reverse chronological order. Listing changes, additions/deletions etc. .
 
 ## Question 13
 
 What does it meant to "push" changes to Github?
 
-You send the changes of your local repository to Github
+Updates the remote branch with local commits.
 
 ## Question 14
 
 What is Github?
 
-A place for us to store and share our code.
+A version control system. It manages the evolution of a set of files (repository) in a highly structured, reversed chronological way. 
 
 ## Question 15
 
 What does it mean to "Commit your changes"?
 
-Send you code to Github
+Create a "snapshot" of your files at a given time. In essence a "save" point. 
```

Comment:
Staging snapshots your changes, commit is adding these staged files/snapshot files to the stack of previous commits